cmd/autoheal/healer.go

```diff
@@ -200,6 +208,52 @@ func (h *Healer) Run(stopCh <-chan struct{}) error {
 	return nil
 }
 
+// onChangeEvent reload config files and
+// refresh rules in rules cache (by sending "Deleted" + "Added" to queue).
+//
+func (h *Healer) onChangeEvent(e *config.ChangeEvent) {
+	mutex := &sync.Mutex{}
+
+	mutex.Lock()
+
+	glog.Infof("Received a config modification event: %s", e.Info)
+	h.loader.Load()
```

Comment:
I still think that when this function is called the configuration should have been reloaded already. This method should just need what is needed to apply that new configuration, which in this case is only reloading the rules, if I understand correctly.

Reply:
I do not know how to trigger the listeners in order, in a nice way  ...

My problem:
On event we need:
1. loader.load
2. reload rules cache

Solutions I thought about:
a. we can register two listeners: but we do not have a mechanism to ensure one will run after the other.
b. we can add a new observer to the loader, that will emit "reload rules cache" once loader.load is finished: but I think that's over engineering it ...